client/lib/url/omit-url-params.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { URL as URLString } from 'types';
+import { Falsy } from 'utility-types';
+
+/**
+ * Internal dependencies
+ */
+import { determineUrlType, URL_TYPE } from './url-type';
+import format from './format';
+
+const BASE_URL = 'http://__domain__.invalid';
+
+/**
+ * Removes given params from a url.
+ *
+ * @param  url URL to be cleaned
+ * @param  paramsToOmit The collection of params or single param to reject
+ * @return Url less the omitted params.
+ */
+export default function omitUrlParams( url: Falsy, paramsToOmit: string | string[] ): null;
+export default function omitUrlParams( url: URLString, paramsToOmit: string | string[] ): URLString;
+export default function omitUrlParams(
+	url: URLString | Falsy,
+	paramsToOmit: string | string[]
+): URLString | null {
+	if ( ! url ) {
+		// Note that we return null for the valid empty string URL.
+		// This is done to maintain compatibility with the previous implementation.
+		return null;
+	}
+
+	const urlType = determineUrlType( url );
+
+	if ( urlType === URL_TYPE.INVALID ) {
+		return null;
+	}
+
+	let toOmit: string[];
+
+	if ( typeof paramsToOmit === 'string' ) {
+		toOmit = [ paramsToOmit ];
+	} else {
+		toOmit = paramsToOmit;
+	}
```

Comment:
I tend to prefer const + ternary for this sort of thing, but up to you:

```ts
const toOmit: string[] = typeof paramsToOmit === 'string'
	? [ paramsToOmit ]
	: paramsToOmit;
```

Reply:
Good idea! Done.